tensorflow_recommenders_addons/dynamic_embedding/python/ops/dynamic_embedding_creator.py

```diff
@@ -225,3 +243,150 @@ def get_config(self):
         'device': self.device,
     }
     return config
+
+
+# SaveableObjectCreator
+class SaveableObjectCreator(object, metaclass=ABCMeta):
+  """
+      
+
+    Example usage:
+
+    ```python
+    saveable_config=
+    ```
+  """
+  _upsert_restore = True
+
+  def __init__(self, config=None):
+    self.config = config
+
+  def create_variable_saveable_object(self, variable, name):
+    raise NotImplementedError('create function must be implemented')
+
+  def create_shard_saveable_object(self,
+                                   variable,
+                                   table,
+                                   shard_idx,
+                                   name,
+                                   full_name=""):
+    raise NotImplementedError('create function must be implemented')
+
+  def set_upsert_restore(self, setting):
+    self._upsert_restore = setting
+
+
+class FileSystemSaveableObjectConfig(object):
```

Comment:
Considering the Config class only serves for MPI/HVD, we may name it `DistributedFileSystem`？ If your initial idea is not only for MPI scenarios, may we need to change it to a common name that will benefit extending in the future?

Reply:
This FileSystem class not only for MPI/HVD, it is also able to be used by TF origin training code. It make user easy to use filesystem for storing DE KV without any additional code by patching TF.